generator/geometry_holder.hpp

```diff
@@ -268,23 +261,32 @@ class GeometryHolder
     return true;
   }
 
-  void FillInnerPointsMask(Points const & points, uint32_t scaleIndex)
+  void FillInnerPoints(Points const & points, uint32_t scaleIndex)
   {
     auto const & src = m_buffer.m_innerPts;
-    CHECK(!src.empty(), ());
+
+    if (src.empty())
+    {
+      m_buffer.m_innerPts = points;
+      // Init the simplification mask to the scaleIndex.
+      // First and last points are present always, hence not stored in the mask.
+      for (size_t i = 1; i < points.size() - 1; ++i)
+        m_buffer.m_ptsSimpMask |= (scaleIndex << (2 * (i - 1)));
+      return;
+    }
```

Comment:
I don't remember, is there a reason to fill this mask too?

Reply:
ATM there is an exceptional case: if only geom3 exists then the mask == 0 (meaning all points are good for geom0-3). This behaviour is illogical, unobvious and relies on visibility index limiting feature display to geom3 only (which actually could be overriden in designer tool, so might cause bugs there).
So its better to keep the mask always filled with proper logical geom levels.

Reply:
Make sense.